assets/js/modules/analytics/datastore/report.js

```diff
@@ -0,0 +1,144 @@
+/**
+ * modules/analytics data store: report.
+ *
+ * Site Kit by Google, Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * External dependencies
+ */
+import invariant from 'invariant';
+import isPlainObject from 'lodash/isPlainObject';
+
+/**
+ * Internal dependencies
+ */
+import API from 'googlesitekit-api';
+import Data from 'googlesitekit-data';
+import { STORE_NAME } from './constants';
+import { stringifyObject } from '../../../util';
+import { createFetchStore } from '../../../googlesitekit/data/create-fetch-store';
+import { isValidDateRange, isValidOrders } from '../../../util/report-validation';
+import { isValidDimensions, isValidMetrics } from '../util/report-validation';
+
+const fetchGetReportStore = createFetchStore( {
+	baseName: 'getReport',
+	controlCallback: ( { options } ) => {
+		return API.get( 'modules', 'analytics', 'report', options );
+	},
+	reducerCallback: ( state, report, { options } ) => {
+		return {
+			...state,
+			reports: {
+				...state.reports,
+				[ stringifyObject( options ) ]: report,
+			},
+		};
+	},
+	argsToParams: ( options ) => {
+		const {
+			metrics,
+			dimensions,
+			orderby,
+		} = options;
```

Comment:
This needs to happen after the initial `invariant` check I think? Otherwise we're destructuring potentially `undefined`.

Reply:
Good catch! Updated.